### PR TITLE
Task_10_NGRX: add store

### PR DESCRIPTION
**Pull request by Volodymyrprodyus**

### CODE REVIEW

File: my-app/src/app/app.component.ts [hunk 1]
```diff
@@ -1,10 +1,32 @@
...
+  public title = 'my-app';
+
+  private subscriptions: Subscription = new Subscription();
+
+  constructor(private contextStoreFacadeService: ContextStoreFacadeService) {}
+
+  public ngOnInit(): void {
+    this.contextStoreFacadeService.dispatchFetchCoursesList();
+
+    this.subscriptions.add(
```
Comment (SteveRusin) [inline my-app/src/app/app.component.ts:20]: Don't forget to unsubscribe on component destroy. 
Also, prefer HostListener as it manages attaching and detaching of the listener itself
@HostListener('window:online')

File: my-app/src/app/courses-page/courses-page.component.ts [hunk 2]
```diff
@@ -16,40 +14,35 @@ import { CoursesService } from './services';
...
-        this.courses$ = this.searchPhrase 
-          ? this.coursesService.getSearchedCourses(this.searchPhrase) 
-          : this.coursesService.getCoursesList();
-      },
-      (err) => console.error(err),
-    )
+    if (Boolean(course.id)) {
+      this.contextStoreFacadeService.dispatchDeleteCourse({ ...course });
+    }
+    this.contextStoreFacadeService.dispatchFetchCoursesList();
```
Comment (SteveRusin) [inline my-app/src/app/courses-page/courses-page.component.ts]: why do you dispatch this? Use select to have live stream of courses and update the list in the effect

File: my-app/src/app/courses-page/courses-page.component.ts [hunk 3]
```diff
@@ -16,40 +14,35 @@ import { CoursesService } from './services';
...
   onLoadMoreCourses(): void {
-    console.log('Load more courses');
-    this.courses$ = this.coursesService.loadMoreCourses();
+    this.contextStoreFacadeService.dispatchLoadMoreCourses();
   }
 
   onCourseSearch(searchPhrase: string): void {
-    this.searchPhrase = searchPhrase;
-    this.courses$ = this.coursesService.getSearchedCourses(searchPhrase);
+    this.contextStoreFacadeService.dispatchSearchCourses(searchPhrase);
```
Comment (SteveRusin) [inline my-app/src/app/courses-page/courses-page.component.ts:35]: you don't need to dispatch dispatchSearchCourses at all if searchPhrase === '' is true

File: my-app/src/app/app.component.ts (hunk 1 above)
Comment (Volodymyrprodyus) [reply my-app/src/app/app.component.ts:20]: fixed

File: my-app/src/app/courses-page/courses-page.component.ts (hunk 3 above)
Comment (Volodymyrprodyus) [reply my-app/src/app/courses-page/courses-page.component.ts:35]: fixed

File: my-app/src/app/courses-page/courses-page.component.ts (hunk 2 above)
Comment (Volodymyrprodyus) [reply my-app/src/app/courses-page/courses-page.component.ts]: fixed

**Pull request merged**